### ISSUE TITLE
Load model with different args than the ones used in training

### ISSUE THREAD

**Issue by Lysimachos** [ThilinaRajapakse/simpletransformers#231]
Anyone knows what happens when we get in a situation like this.

Looks like when I loaded a model with different args than the ones used in training in multi-label classification the model predicted 0 for all labels for all instances.

How can I fix this?

**Comment by ThilinaRajapakse** [comment]
What are the args that you changed?

**Comment by Lysimachos** [comment]
The following args:
max_seq_length
process_count
gradient_accumulation_steps
train_batch_size

The problem I am facing now is that even if I load a "clean" model and train it (fine-tuning) from scratch the model still predicts 0 for all labels for all instances. I think there is something to do with the cache and generally how it is saved. Is there any information about models and features saving and loading?

Thanks again Thilina for your interest.

**Comment by ThilinaRajapakse** [comment]
I think changing the max_seq_length will break the model.

The [readme](https://github.com/ThilinaRajapakse/simpletransformers#default-settings) should contain all the info.

**Comment by Lysimachos** [comment]
Yes I have seen this readme but I haven't figured out how old runs are connected with new model trainings.  My problem is that although I have downloaded the model from source (using force download mode), train it from start ( I see the loss function that it is printed getting smaller and smaller) and even get good results in eval sample (eval loss is around the same with the loss last printed on training process) the prediction in the test sample are wrong (again full of 0). 

Is there any information regarding the files: runs, outputs and cached_dir that are created. Is anything else written -saved when training or loading?

**Comment by ThilinaRajapakse** [comment]
- `runs` contains tensorboard files
- `outputs` contains saved models
- `cached_dir` contains cache files like converted features

Are you loading the fine tuned model with the path to the trained model as below?

`model = ClassificationModel("roberta", "outputs")`

**Comment by Lysimachos** [comment]
Yes I loaded the trained model like this:
model = ClassificationModel("roberta", "outputs", args)

Maybe the way I expressed my issue was misleading. What I am saying is that after breaking the model by loading a model with different args than the ones used in training the code also broke. New models offer terrible predictions after training. Since no changes have been made in the code nor to the system I am guessing that there is something with the files written somewhere.
